WIP: qt5.qtwebengine: Fix darwin build

knedlsepp opened the pull request:
###### Motivation for this change
This fixes issue: #36932

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

knedlsepp committed:
qt5.qtwebengine: Prevent falsely succeeding build

This prevents the build falsely succeeding, despite generating no files.
See e.g. https://hydra.nixos.org/build/73282535

knedlsepp committed:
qt5.qtwebengine: Fix darwin build

The build failed with the following error before:

```
Could not resolve SDK product version for 'macosx'.
Building Qt WebEngine requires a macOS SDK version of 10.11 or newer.
Current version is .
QtWebEngine will not be built.
```

knedlsepp committed:
qt5.qtwebengine: Fix darwin build

The build failed with the following error before:

```
Could not resolve SDK product version for 'macosx'.
Building Qt WebEngine requires a macOS SDK version of 10.11 or newer.
Current version is .
QtWebEngine will not be built.
```

knedlsepp commented:
@matthewbauer Tried to fix qtwebengine on darwin. It's still compiling, but past the previous problems, so I'm optimistic this builds.

Mic92 commented on pkgs/development/libraries/qt-5/modules/qtwebengine.nix:
```diff
@@ -24,8 +25,9 @@ qtModule {
   qtInputs = [ qtdeclarative qtquickcontrols qtlocation qtwebchannel ];
   nativeBuildInputs = [
-    bison coreutils flex git gperf ninja pkgconfig python2 which gn
+    bison coreutils flex git gperf ninja pkgconfig python2 which gn xcbuild
```
Is it also required on Linux?

knedlsepp replied:
Whoops. No it isn't.

knedlsepp replied:
Fixed. Thanks for the review.

knedlsepp commented on pkgs/development/libraries/qt-5/modules/qtwebengine.nix:
```diff
@@ -176,6 +183,12 @@ EOF
     Prefix = ..
     EOF
     paxmark m $out/libexec/QtWebEngineProcess
+  '' + ''
```
@Mic92 What do you think about this? Should it also be darwin only? Currently the build succeeds incorrectly on darwin, but it might also on linux: https://hydra.nixos.org/build/73282201/log

Mic92 replied:
Should be ok.

Mic92 commented:
Did it build in the end?

knedlsepp commented:
I couldn't finish the build yesterday. (Linux build https://hydra.nixos.org/build/73282040 took over 9 hours). I will only be able to resume the build tonight.

knedlsepp commented:
It actually didn't go through.
```
[2375/16039] CXX obj/sandbox/mac/sandbox/bootstrap_sandbox.o
clang-5.0: warning: no such sysroot directory: '../../3rdparty/chromium/build/config/mac/10.10.0' [-Wmissing-sysroot]
[2376/16039] CXX obj/sandbox/mac/sandbox/mach_message_server.o
FAILED: obj/sandbox/mac/sandbox/mach_message_server.o
../../../../../../../nix/store/8b0gj3bmcy5bh1zcnhn0146199b3m5kb-clang-wrapper-5.0.1/bin/clang++ -MMD -MF obj/sandbox/mac/sandbox/mach_message_server.o.d -DSANDBOX_IMPLEMENTATION -DV8_DEPRECATION_WARNINGS -DNO_TCMALLOC -DDISABLE_NACL -DFULL_SAFE_BROWSING -DSAFE_BROWSING_CSD -DSAFE_BROWSING_DB_LOCAL -DCHROMIUM_BUILD -DFIELDTRIAL_TESTING_ENABLED -DTOOLKIT_QT -DCR_CLANG_REVISION=\"307486-1\" -DCR_XCODE_VERSION=9.1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORE=0 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -Igen -I../../3rdparty/chromium -fno-strict-aliasing -fstack-protector -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -Wno-unknown-attributes -Wno-unknown-warning-option -fcolor-diagnostics -arch x86_64 -Wall -Wextra -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Wunguarded-availability -Wno-missing-field-initializers -Wno-unused-parameter -Wno-c++11-narrowing -Wno-covered-switch-default -Wno-unneeded-internal-declaration -Wno-inconsistent-missing-override -Wno-undefined-var-template -Wno-nonportable-include-path -Wno-address-of-packed-member -Wno-unused-lambda-capture -Wno-user-defined-warnings -O2 -fno-omit-frame-pointer -g0 -isysroot ../../3rdparty/chromium/build/config/mac/10.10.0 -mmacosx-version-min=10.10 -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Wno-parentheses-equality -Wno-tautological-compare -Wno-undefined-bool-conversion -Wno-tautological-undefined-compare -std=c++11 -stdlib=libc++ -fno-rtti -fno-exceptions -fvisibility-inlines-hidden -c ../../3rdparty/chromium/sandbox/mac/mach_message_server.cc -o obj/sandbox/mac/sandbox/mach_message_server.o
clang-5.0: warning: no such sysroot directory: '../../3rdparty/chromium/build/config/mac/10.10.0' [-Wmissing-sysroot]
../../3rdparty/chromium/sandbox/mac/mach_message_server.cc:91:22: error: use of undeclared identifier 'audit_token_to_pid'
  pid_t sender_pid = audit_token_to_pid(trailer->msgh_audit);
                     ^
1 error generated.
ninja: build stopped: subcommand failed.
```

matthewbauer committed:
qtwebengine: fixup openbsm issue

Chromium’s sandbox uses some proprietary apple features. I’ve added a
CPP function to substitute the relevant info.

matthewbauer commented:
Ok well that's getting closer. It's an issue with Chromium using Apple's proprietary OpenBSM. I just pushed a commit that hopefully fixes it.

knedlsepp commented:
How did you come up with that?

matthewbauer commented:
From this: https://github.com/darlinghq/darling-bsm/commit/a0c57822d493c9a387b8763e4359ffab606bb090

I have no idea if it's right but it compiles!

matthewbauer commented:
@GrahamcOfBorg build qt5.qtwebengine

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40149&attempt_id=7c218aca-6757-4d72-935e-623f3dac45e8)

Attempted: qt5.qtwebengine

<details><summary>Partial log (click to expand)</summary><p>

```
Building gn manually in a temporary directory for bootstrapping...
Building gn using itself to out/Release...
Project ERROR: GN build error!
make[2]: *** [Makefile:60: sub-gn-pro-make_first] Error 3
make[2]: Leaving directory '/private/tmp/nix-build-qtwebengine-5.10.1.drv-0/qtwebengine-everywhere-src-5.10.1/src/buildtools'
make[1]: *** [Makefile:54: sub-buildtools-make_first] Error 2
make[1]: Leaving directory '/private/tmp/nix-build-qtwebengine-5.10.1.drv-0/qtwebengine-everywhere-src-5.10.1/src'
make: *** [Makefile:47: sub-src-make_first] Error 2
builder for '/nix/store/9sjbfm1blrhj6k50i673cabq8ljzfq9r-qtwebengine-5.10.1.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/9sjbfm1blrhj6k50i673cabq8ljzfq9r-qtwebengine-5.10.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40149&attempt_id=a424141e-0252-44ba-bbab-82aef13d1cbb)

Attempted: qt5.qtwebengine

<details><summary>Partial log (click to expand)</summary><p>

```
[4846/17529] CXX obj/third_party/icu/bundled_icuuc/sharedobject.o
[4847/17529] CXX obj/third_party/icu/bundled_icui18n/ucol.o
[4848/17529] CXX obj/third_party/icu/bundled_icui18n/tmutamt.o
[4849/17529] CXX obj/third_party/icu/bundled_icuuc/stringtriebuilder.o
[4850/17529] CXX obj/third_party/icu/bundled_icuuc/cstr.o
[4851/17529] CXX obj/third_party/icu/bundled_icuuc/dictionarydata.o
[4852/17529] CXX obj/third_party/icu/bundled_icuuc/ucol_swp.o
[4853/17529] CXX obj/third_party/icu/bundled_icuuc/ubidiln.o
building of '/nix/store/spbf5p9f8kg05rkdihn97qypx6pz4mr1-qtwebengine-5.10.1.drv' timed out after 3600 seconds
[31;1merror:[0m build of '/nix/store/spbf5p9f8kg05rkdihn97qypx6pz4mr1-qtwebengine-5.10.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40149&attempt_id=835bebf0-aab1-450e-aba4-5dbcaeb495f8)

Attempted: qt5.qtwebengine

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/bin/moc -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_UI_DELEGATES -DENABLE_SPELLCHECK -DENABLE_PRINTING -DENABLE_PDF -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_EXCEPTIONS -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -DQT_WEBENGINECORE_LIB -DQT_QUICK_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_WEBCHANNEL_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_POSITIONING_LIB -DQT_CORE_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB --include .moc/moc_predefs.h -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/mkspecs/linux-g++ -I/build/qtwebengine-everywhere-src-5.10.1/src/webenginewidgets -I/build/qtwebengine-everywhere-src-5.10.1/src/webenginewidgets -I/build/qtwebengine-everywhere-src-5.10.1/src/webenginewidgets/api -I/build/qtwebengine-everywhere-src-5.10.1/src/core -I/build/qtwebengine-everywhere-src-5.10.1/src/core/api -I/build/qtwebengine-everywhere-src-5.10.1/src/webengine/api -I/build/qtwebengine-everywhere-src-5.10.1/include -I/build/qtwebengine-everywhere-src-5.10.1/include/QtWebEngineWidgets -I/build/qtwebengine-everywhere-src-5.10.1/include/QtWebEngineWidgets/5.10.1 -I/build/qtwebengine-everywhere-src-5.10.1/include/QtWebEngineWidgets/5.10.1/QtWebEngineWidgets -I/build/qtwebengine-everywhere-src-5.10.1/include/QtWebEngineCore -I/nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtPrintSupport -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui -I/nix/store/hw8x3ygmrhd0z9n3ydwqkn953qyjaw7i-qtwebchannel-5.10.1-dev/include/QtWebChannel -I/nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtNetwork -I/nix/store/h8ykj5fykb4a98ljzfmgi1mf2adk6ix3-qtlocation-5.10.1-dev/include/QtPositioning -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore -I/nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1 -I/nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1/QtQuick -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1 -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1/QtWidgets -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1 -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1/QtGui -I/nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1 -I/nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1/QtQml -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1 -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1/QtCore -I/nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuickWidgets -I/nix/store/q1z276lzf9jcdv23s7mpma3rrcly9jrq-flex-2.6.4/include -I/nix/store/l9j2jsc9flrbmpf799nw9wdq59gpwms8-python-2.7.14/include -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include -I/nix/store/jlk47c1q7zg0ivb6k8hvcwq3jidz4skj-libxml2-2.9.7-dev/include -I/nix/store/vs091h9m987lcdqili0qmxchk1pw5329-zlib-1.2.11-dev/include -I/nix/store/8nqmk6qh4f7glnqg0k7zrp2zm8h72c0g-libxslt-1.1.32-dev/include -I/nix/store/vp2imwdhvqfi2l2d39zqhqz7y56aw2kl-openssl-1.0.2o-dev/include -I/nix/store/jc4qfpnwc5pngzcdmczyka31silfb86k-sqlite-3.22.0-dev/include -I/nix/store/4dqcjmv155miffs9qasdli0ck8sndy57-harfbuzz-icu-1.7.6-dev/include -I/nix/store/xwdjw18l62h2dvq814hrrvdv6g2na0y6-graphite2-1.3.6/include -I/nix/store/ak2d8cqlzd646xya5dnamr22n8zh38q6-icu4c-59.1-dev/include -I/nix/store/2ck6fr449hby8haj89wnjh6zs5y2k7a6-harfbuzz-1.7.6-dev/include -I/nix/store/n0q4nyl8sw0dbk5yi8kiyvrrf84blzn0-libjpeg-turbo-1.5.3-dev/include -I/nix/store/v9cz97wzqaw1rfwwj4a9fm7mmx9g3kx7-libpng-apng-1.6.34-dev/include -I/nix/store/ha19wzfcfgy59kf9syhwwzwl20ilcr9d-libtiff-4.0.9-dev/include -I/nix/store/dc23kf0m4idzci733x5f4n94bvhzfha6-xz-5.2.3-dev/include -I/nix/store/hnvkcd8pf7nmlbh4dgin44zka3g3vl5w-pcre2-10.23-dev/include -I/nix/store/s0qic8ah9j8qxk8yh1m3xpzcjaz5b8pw-dbus-1.12.6-dev/include -I/nix/store/dpydsfzmxm2dfmsr1c2llvrra9y663qx-expat-2.2.5-dev/include -I/nix/store/szcis3117qd65i22322fjsk3bzljc6bc-glib-2.56.0-dev/include -I/nix/store/fg6asqlsnc0wnv9fj6zp29w9aadbm6ir-libffi-3.2.1-dev/include -I/nix/store/biv22ang6z5yviggwb0v2yydvsf82mgc-gettext-0.19.8/include -I/nix/store/rk71nxcvkrlr1bavp7zic074w5blzc9i-glibc-iconv-2.27/include -I/nix/store/a32dmays1bbjswd63rzw4mqrbwbwzxn2-systemd-238-dev/include -I/nix/store/cywh34sn8b00qil2g112bdhgdklkh2ca-fontconfig-2.12.6-dev/include -I/nix/store/11mvvhykc4svx4f5p5ssw5l87h7v99q6-freetype-2.9-dev/include -I/nix/store/6kblqz81diyg5r09x3qdf4b9x7ki02mx-bzip2-1.0.6.0.1-dev/include -I/nix/store/38fijq0yaj57d165sld8vx9kk1ns727h-libX11-1.6.5-dev/include -I/nix/store/gxl4fhz1j2q2y0h0k5hl32c5qccx9wn2-libxcb-1.12-dev/include -I/nix/store/lrc50kyzy9y28b6dxpn4caxi2jn67jrn-xproto-7.0.31/include -I/nix/store/zkapppbw3hr7jmmppd4miqfx8xpky7mf-kbproto-1.0.7/include -I/nix/store/nf8dswgf2dg3x01fb7k7k2q823z6kmgh-libXcomposite-0.4.4-dev/include -I/nix/store/p1hxydbsivlkzankq5da2jq007z3icgg-libXfixes-5.0.3-dev/include -I/nix/store/lz6yw7cazv7jqb7w976sh6s5i4msxqrk-fixesproto-5.0/include -I/nix/store/42mna93w3s2zjpjk6m32gij2pjb2yw2a-xextproto-7.3.0/include -I/nix/store/843z48hmp1hrbzjnrbajb5rlzqszgw6r-compositeproto-0.4.2/include -I/nix/store/jd4fkighib2q4smrg279gxkfl0d18z8s-libXext-1.3.3-dev/include -I/nix/store/ybjh7f394hk0bi1mcyac73i42kk0x5ni-libXau-1.0.8-dev/include -I/nix/store/kidr7p772pmkr1mpwsk6f89qm1igh32x-libXi-1.7.9-dev/include -I/nix/store/y34q1nyrbl369q9cdj4zhakzpf3557rh-inputproto-2.3.2/include -I/nix/store/g7w9sd4jldcca5y25dyxa9yq7f465jd0-libXrender-0.9.10-dev/include -I/nix/store/0f0sbxl04030hjiww7y8jvzpxlf0r6ms-renderproto-0.11.1/include -I/nix/store/k4lbpl0ym2yrn2drjvrwkfkrdr8iyx4m-libxkbcommon-0.8.0-dev/include -I/nix/store/x1l66yzq180j6j3w0rvsi11kavcv7d25-xcb-util-0.4.0-dev/include -I/nix/store/41pqn4l9bvrd7b5mlsc22286imsiv8m8-xcb-util-image-0.4.0-dev/include -I/nix/store/5k84wlla5wqxhrn4frc9mkr7w3j7bmlm-xcb-util-keysyms-0.4.0-dev/include -I/nix/store/dfycb8309l36wy5b211wwrf9x0l1ihx8-xcb-util-renderutil-0.3.9-dev/include -I/nix/store/8ml65yc96jj2ac1q98ynk89rpdd84kpj-xcb-util-wm-0.4.1-dev/include -I/nix/store/nn8y102z124cp2ijcjmz72j53b7ibg5y-libGL-1.0.0-dev/include -I/nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include -I/nix/store/61g5zxi86qrhc4x4pc0fwa30qwixbxsl-qtsvg-5.10.1-dev/include -I/nix/store/cxz7bjh29wwr50jrhmwx89ahmiild184-qtxmlpatterns-5.10.1-dev/include -I/nix/store/h8ykj5fykb4a98ljzfmgi1mf2adk6ix3-qtlocation-5.10.1-dev/include -I/nix/store/0n5sr4i0ib0mknsi258fmigqjm55s66c-qtmultimedia-5.10.1-dev/include -I/nix/store/hw8x3ygmrhd0z9n3ydwqkn953qyjaw7i-qtwebchannel-5.10.1-dev/include -I/nix/store/2xs4cz5zkp3blf48njvxx1xy37rjyd77-libwebp-0.6.1/include -I/nix/store/d7ss2hxaj6xmb9kl7v4kcwgjfcp0ramh-libsrtp-2.1.0/include -I/nix/store/hy5jf061b9wxws6wijn2qjcraa569w0l-libpcap-1.8.1/include -I/nix/store/iaq951wj5caydxdkp80899azbcmdij5p-libvpx-1.6.1-dev/include -I/nix/store/jhwfcnd7ya4hhrddi06kir5k9j78ag1m-libopus-1.2.1-dev/include -I/nix/store/kzizr56mh6cjq8bfyqdlv2glcyjl8lv1-libevent-2.1.8-dev/include -I/nix/store/plll3ixfafm27dbdzm65qklm858rwb8i-minizip-1.2.11/include -I/nix/store/q9l997ccl6fiaym6yl451jayc79gk4nz-snappy-1.1.7-dev/include -I/nix/store/202a8vcf2j9dk7skamqwz6ms75n4np3a-nss-3.35-dev/include -I/nix/store/43i29r83lkmx4h603lkykd05sxmailax-nspr-4.18-dev/include -I/nix/store/k52s6140w4wdcsk7w0jwx46rdckp452j-protobuf-3.4.1/include -I/nix/store/5zy7dbnbjck04nyyhd5fyhs0y1r0gz0i-jsoncpp-1.8.4/include -I/nix/store/vcp8lq2jb69q8aq055mbyfz5s12y3r8r-alsa-lib-1.1.6-dev/include -I/nix/store/zni83wa8f2srvpak6swqf266cn6jz9sr-libcap-2.25-dev/include -I/nix/store/lvx2yh4i7map0yrc6sm7apk2ncb0l94d-attr-2.4.47-dev/include -I/nix/store/myn1xcjg0yavvrwgx364ky55q7r96g21-pciutils-3.5.6/include -I/nix/store/g92ns7x3fbn2d0s3gbvgm521yg31x9c6-libXScrnSaver-1.2.2/include -I/nix/store/2d8waz3rxq7niwikf4jgnr577ldpsdyz-scrnsaverproto-1.2.2/include -I/nix/store/k31vpnxlzhx770lgymqj6a0pb449s77h-libXcursor-1.1.15-dev/include -I/nix/store/9d0nnpgcdnnvblizxbjxzpjacl9q7ysf-libXrandr-1.5.1-dev/include -I/nix/store/bg4r4kh6x4idc45p9qwv4kqb28q24clx-randrproto-1.5.0/include -I/nix/store/i7h12qghg3hcgir2yzg44b44fzcsbh8v-libpciaccess-0.14/include -I/nix/store/r967ycbx4rilxg7kyjm1pigyqss959bp-libXtst-1.2.3/include -I/nix/store/wpg01z92pk4zsfqm46ra6zzhdh8higqk-recordproto-1.14.2/include -I/nix/store/v2rqr44jk0kpmhvssdbycgrnv5356a2s-libXdamage-1.1.4-dev/include -I/nix/store/qj655rjlw0qb9fa178xk6mp5maa107cm-damageproto-1.2.1/include -I/nix/store/q3nmibrfv1r2spjywkyy9r0wj2cy8a6k-libdrm-2.4.91-dev/include -I/nix/store/ng5kbr6kmh979qaachifbsbblydrcmqc-gcc-7.3.0/include/c++/7.3.0 -I/nix/store/ng5kbr6kmh979qaachifbsbblydrcmqc-gcc-7.3.0/include/c++/7.3.0/x86_64-unknown-linux-gnu -I/nix/store/ng5kbr6kmh979qaachifbsbblydrcmqc-gcc-7.3.0/include/c++/7.3.0/backward -I/nix/store/ng5kbr6kmh979qaachifbsbblydrcmqc-gcc-7.3.0/lib/gcc/x86_64-unknown-linux-gnu/7.3.0/include -I/nix/store/ng5kbr6kmh979qaachifbsbblydrcmqc-gcc-7.3.0/include -I/nix/store/ng5kbr6kmh979qaachifbsbblydrcmqc-gcc-7.3.0/lib/gcc/x86_64-unknown-linux-gnu/7.3.0/include-fixed -I/nix/store/lb3kkvyylnbcxn5jad9i11szxlhkpkjq-glibc-2.27-dev/include api/qwebenginepage.h -o .moc/moc_qwebenginepage.cpp
g++ -c -include .pch/Qt5WebEngineWidgets -pipe -O2 -std=c++1z -fvisibility=hidden -fvisibility-inlines-hidden -fno-exceptions -Wall -W -Wvla -Wdate-time -Wshift-overflow=2 -Wduplicated-cond -Wno-stringop-overflow -D_REENTRANT -fPIC -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_UI_DELEGATES -DENABLE_SPELLCHECK -DENABLE_PRINTING -DENABLE_PDF -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_EXCEPTIONS -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -DQT_WEBENGINECORE_LIB -DQT_QUICK_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_WEBCHANNEL_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_POSITIONING_LIB -DQT_CORE_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I. -Iapi -I../core -I../core/api -I../webengine/api -I../../include -I../../include/QtWebEngineWidgets -I../../include/QtWebEngineWidgets/5.10.1 -I../../include/QtWebEngineWidgets/5.10.1/QtWebEngineWidgets -I../../include/QtWebEngineCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtPrintSupport -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui -isystem /nix/store/hw8x3ygmrhd0z9n3ydwqkn953qyjaw7i-qtwebchannel-5.10.1-dev/include/QtWebChannel -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtNetwork -isystem /nix/store/h8ykj5fykb4a98ljzfmgi1mf2adk6ix3-qtlocation-5.10.1-dev/include/QtPositioning -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1/QtGui -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuickWidgets -I.moc -I/nix/store/nz47k470m1gppw0rbmjd3vs5i9p29qv5-mesa-noglu-17.3.8-dev/include -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/mkspecs/linux-g++ -o .obj/qwebengineview.o api/qwebengineview.cpp
g++ -c -include .pch/Qt5WebEngineWidgets -pipe -O2 -std=c++1z -fvisibility=hidden -fvisibility-inlines-hidden -fno-exceptions -Wall -W -Wvla -Wdate-time -Wshift-overflow=2 -Wduplicated-cond -Wno-stringop-overflow -D_REENTRANT -fPIC -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_UI_DELEGATES -DENABLE_SPELLCHECK -DENABLE_PRINTING -DENABLE_PDF -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_EXCEPTIONS -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -DQT_WEBENGINECORE_LIB -DQT_QUICK_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_WEBCHANNEL_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_POSITIONING_LIB -DQT_CORE_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I. -Iapi -I../core -I../core/api -I../webengine/api -I../../include -I../../include/QtWebEngineWidgets -I../../include/QtWebEngineWidgets/5.10.1 -I../../include/QtWebEngineWidgets/5.10.1/QtWebEngineWidgets -I../../include/QtWebEngineCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtPrintSupport -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui -isystem /nix/store/hw8x3ygmrhd0z9n3ydwqkn953qyjaw7i-qtwebchannel-5.10.1-dev/include/QtWebChannel -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtNetwork -isystem /nix/store/h8ykj5fykb4a98ljzfmgi1mf2adk6ix3-qtlocation-5.10.1-dev/include/QtPositioning -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1/QtGui -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuickWidgets -I.moc -I/nix/store/nz47k470m1gppw0rbmjd3vs5i9p29qv5-mesa-noglu-17.3.8-dev/include -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/mkspecs/linux-g++ -o .obj/moc_qwebenginedownloaditem.o .moc/moc_qwebenginedownloaditem.cpp
g++ -c -include .pch/Qt5WebEngineWidgets -pipe -O2 -std=c++1z -fvisibility=hidden -fvisibility-inlines-hidden -fno-exceptions -Wall -W -Wvla -Wdate-time -Wshift-overflow=2 -Wduplicated-cond -Wno-stringop-overflow -D_REENTRANT -fPIC -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_UI_DELEGATES -DENABLE_SPELLCHECK -DENABLE_PRINTING -DENABLE_PDF -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_EXCEPTIONS -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -DQT_WEBENGINECORE_LIB -DQT_QUICK_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_WEBCHANNEL_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_POSITIONING_LIB -DQT_CORE_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I. -Iapi -I../core -I../core/api -I../webengine/api -I../../include -I../../include/QtWebEngineWidgets -I../../include/QtWebEngineWidgets/5.10.1 -I../../include/QtWebEngineWidgets/5.10.1/QtWebEngineWidgets -I../../include/QtWebEngineCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtPrintSupport -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui -isystem /nix/store/hw8x3ygmrhd0z9n3ydwqkn953qyjaw7i-qtwebchannel-5.10.1-dev/include/QtWebChannel -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtNetwork -isystem /nix/store/h8ykj5fykb4a98ljzfmgi1mf2adk6ix3-qtlocation-5.10.1-dev/include/QtPositioning -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1/QtGui -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuickWidgets -I.moc -I/nix/store/nz47k470m1gppw0rbmjd3vs5i9p29qv5-mesa-noglu-17.3.8-dev/include -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/mkspecs/linux-g++ -o .obj/moc_qwebenginefullscreenrequest.o .moc/moc_qwebenginefullscreenrequest.cpp
g++ -c -include .pch/Qt5WebEngineWidgets -pipe -O2 -std=c++1z -fvisibility=hidden -fvisibility-inlines-hidden -fno-exceptions -Wall -W -Wvla -Wdate-time -Wshift-overflow=2 -Wduplicated-cond -Wno-stringop-overflow -D_REENTRANT -fPIC -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_UI_DELEGATES -DENABLE_SPELLCHECK -DENABLE_PRINTING -DENABLE_PDF -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_EXCEPTIONS -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -DQT_WEBENGINECORE_LIB -DQT_QUICK_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_WEBCHANNEL_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_POSITIONING_LIB -DQT_CORE_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I. -Iapi -I../core -I../core/api -I../webengine/api -I../../include -I../../include/QtWebEngineWidgets -I../../include/QtWebEngineWidgets/5.10.1 -I../../include/QtWebEngineWidgets/5.10.1/QtWebEngineWidgets -I../../include/QtWebEngineCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtPrintSupport -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui -isystem /nix/store/hw8x3ygmrhd0z9n3ydwqkn953qyjaw7i-qtwebchannel-5.10.1-dev/include/QtWebChannel -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtNetwork -isystem /nix/store/h8ykj5fykb4a98ljzfmgi1mf2adk6ix3-qtlocation-5.10.1-dev/include/QtPositioning -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1/QtGui -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuickWidgets -I.moc -I/nix/store/nz47k470m1gppw0rbmjd3vs5i9p29qv5-mesa-noglu-17.3.8-dev/include -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/mkspecs/linux-g++ -o .obj/moc_qwebengineprofile.o .moc/moc_qwebengineprofile.cpp
g++ -c -include .pch/Qt5WebEngineWidgets -pipe -O2 -std=c++1z -fvisibility=hidden -fvisibility-inlines-hidden -fno-exceptions -Wall -W -Wvla -Wdate-time -Wshift-overflow=2 -Wduplicated-cond -Wno-stringop-overflow -D_REENTRANT -fPIC -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_UI_DELEGATES -DENABLE_SPELLCHECK -DENABLE_PRINTING -DENABLE_PDF -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_EXCEPTIONS -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -DQT_WEBENGINECORE_LIB -DQT_QUICK_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_WEBCHANNEL_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_POSITIONING_LIB -DQT_CORE_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I. -Iapi -I../core -I../core/api -I../webengine/api -I../../include -I../../include/QtWebEngineWidgets -I../../include/QtWebEngineWidgets/5.10.1 -I../../include/QtWebEngineWidgets/5.10.1/QtWebEngineWidgets -I../../include/QtWebEngineCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtPrintSupport -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui -isystem /nix/store/hw8x3ygmrhd0z9n3ydwqkn953qyjaw7i-qtwebchannel-5.10.1-dev/include/QtWebChannel -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtNetwork -isystem /nix/store/h8ykj5fykb4a98ljzfmgi1mf2adk6ix3-qtlocation-5.10.1-dev/include/QtPositioning -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1/QtGui -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuickWidgets -I.moc -I/nix/store/nz47k470m1gppw0rbmjd3vs5i9p29qv5-mesa-noglu-17.3.8-dev/include -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/mkspecs/linux-g++ -o .obj/moc_render_widget_host_view_qt_delegate_widget.o .moc/moc_render_widget_host_view_qt_delegate_widget.cpp
g++ -c -include .pch/Qt5WebEngineWidgets -pipe -O2 -std=c++1z -fvisibility=hidden -fvisibility-inlines-hidden -fno-exceptions -Wall -W -Wvla -Wdate-time -Wshift-overflow=2 -Wduplicated-cond -Wno-stringop-overflow -D_REENTRANT -fPIC -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_UI_DELEGATES -DENABLE_SPELLCHECK -DENABLE_PRINTING -DENABLE_PDF -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_EXCEPTIONS -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -DQT_WEBENGINECORE_LIB -DQT_QUICK_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_WEBCHANNEL_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_POSITIONING_LIB -DQT_CORE_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I. -Iapi -I../core -I../core/api -I../webengine/api -I../../include -I../../include/QtWebEngineWidgets -I../../include/QtWebEngineWidgets/5.10.1 -I../../include/QtWebEngineWidgets/5.10.1/QtWebEngineWidgets -I../../include/QtWebEngineCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtPrintSupport -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui -isystem /nix/store/hw8x3ygmrhd0z9n3ydwqkn953qyjaw7i-qtwebchannel-5.10.1-dev/include/QtWebChannel -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtNetwork -isystem /nix/store/h8ykj5fykb4a98ljzfmgi1mf2adk6ix3-qtlocation-5.10.1-dev/include/QtPositioning -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1/QtGui -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuickWidgets -I.moc -I/nix/store/nz47k470m1gppw0rbmjd3vs5i9p29qv5-mesa-noglu-17.3.8-dev/include -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/mkspecs/linux-g++ -o .obj/moc_messagebubblewidget_p.o .moc/moc_messagebubblewidget_p.cpp
g++ -c -include .pch/Qt5WebEngineWidgets -pipe -O2 -std=c++1z -fvisibility=hidden -fvisibility-inlines-hidden -fno-exceptions -Wall -W -Wvla -Wdate-time -Wshift-overflow=2 -Wduplicated-cond -Wno-stringop-overflow -D_REENTRANT -fPIC -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_UI_DELEGATES -DENABLE_SPELLCHECK -DENABLE_PRINTING -DENABLE_PDF -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_BUILD_WEBENGINEWIDGETS_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_EXCEPTIONS -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -DQT_WEBENGINECORE_LIB -DQT_QUICK_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_WEBCHANNEL_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_POSITIONING_LIB -DQT_CORE_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I. -Iapi -I../core -I../core/api -I../webengine/api -I../../include -I../../include/QtWebEngineWidgets -I../../include/QtWebEngineWidgets/5.10.1 -I../../include/QtWebEngineWidgets/5.10.1/QtWebEngineWidgets -I../../include/QtWebEngineCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtPrintSupport -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui -isystem /nix/store/hw8x3ygmrhd0z9n3ydwqkn953qyjaw7i-qtwebchannel-5.10.1-dev/include/QtWebChannel -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtNetwork -isystem /nix/store/h8ykj5fykb4a98ljzfmgi1mf2adk6ix3-qtlocation-5.10.1-dev/include/QtPositioning -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuick/5.10.1/QtQuick -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtWidgets/5.10.1/QtWidgets -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtGui/5.10.1/QtGui -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1 -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQml/5.10.1/QtQml -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1 -isystem /nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/include/QtCore/5.10.1/QtCore -isystem /nix/store/8vg2w4685b2yqzwz18yw7ic7hpsqp2vy-qtdeclarative-5.10.1-dev/include/QtQuickWidgets -I.moc -I/nix/store/nz47k470m1gppw0rbmjd3vs5i9p29qv5-mesa-noglu-17.3.8-dev/include -I/nix/store/12fzvkzksb0z3qby17kz4g99k8vi6d80-qtbase-5.10.1-dev/mkspecs/linux-g++ -o .obj/qwebenginepage.o api/qwebenginepage.cpp
building of '/nix/store/nh4xmprgx0vsp080jkjr0wlahv2977vp-qtwebengine-5.10.1.drv' timed out after 4000 seconds
error: build of '/nix/store/nh4xmprgx0vsp080jkjr0wlahv2977vp-qtwebengine-5.10.1.drv' failed
```
</p></details>

knedlsepp commented:
Oddly enough it's the same issue again:
```
../../../../base/mac/mach_port_broker.mm:157:21: error: use of undeclared identifier 'audit_token_to_pid'
  pid_t child_pid = audit_token_to_pid(msg.trailer.msgh_audit);
```
Are the `NIX_CFLAGS_COMPILE` flags not respected for Objective-C++?

matthewbauer commented:
Yeah we might need to just go back to patching it. Try reverting my commit & adding one for the file that needs it. Hopefully there are not too many more cases.

matthewbauer commented:
@GrahamcOfBorg build qt5.qtwebengine

Ok patch applied in 79dcdce63387d8b0c624abbc00d00acaa7912a43

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40149&attempt_id=d21b371e-c5b4-431e-a82b-9beee68abbe6)

Attempted: qt5.qtwebengine

<details><summary>Partial log (click to expand)</summary><p>

```
ninja: build stopped: subcommand failed.
make[3]: *** [Makefile.gn_run:345: run_ninja] Error 1
make[3]: Leaving directory '/private/tmp/nix-build-qtwebengine-5.10.1.drv-0/qtwebengine-everywhere-src-5.10.1/src/core'
make[2]: *** [Makefile:80: sub-gn_run-pro-make_first] Error 2
make[2]: Leaving directory '/private/tmp/nix-build-qtwebengine-5.10.1.drv-0/qtwebengine-everywhere-src-5.10.1/src/core'
make[1]: *** [Makefile:79: sub-core-make_first] Error 2
make[1]: Leaving directory '/private/tmp/nix-build-qtwebengine-5.10.1.drv-0/qtwebengine-everywhere-src-5.10.1/src'
make: *** [Makefile:47: sub-src-make_first] Error 2
builder for '/nix/store/ak0k6dzxajg8srpkq5p5ca7fl985k3gi-qtwebengine-5.10.1.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/ak0k6dzxajg8srpkq5p5ca7fl985k3gi-qtwebengine-5.10.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40149&attempt_id=017a28a6-2836-44b8-aa82-6118cc98792e)

Attempted: qt5.qtwebengine

<details><summary>Partial log (click to expand)</summary><p>

```
[9327/17795] CXX obj/v8/v8_builtins_generators/binary-op-assembler.o
[9328/17795] CXX obj/v8/v8_builtins_generators/keyed-store-generic.o
[9329/17795] CXX obj/v8/v8_builtins_generators/setup-interpreter-internal.o
[9330/17795] CXX obj/v8/v8_builtins_generators/interpreter-assembler.o
[9331/17795] STAMP obj/v8/v8_builtins_setup.inputdeps.stamp
[9332/17795] CXX obj/v8/v8_builtins_generators/interpreter-intrinsics-generator.o
[9333/17795] CXX obj/v8/v8_builtins_generators/interpreter-generator.o
[9334/17795] CXX obj/v8/v8_builtins_setup/setup-isolate-full.o
building of '/nix/store/nh4xmprgx0vsp080jkjr0wlahv2977vp-qtwebengine-5.10.1.drv' timed out after 1800 seconds
error: build of '/nix/store/nh4xmprgx0vsp080jkjr0wlahv2977vp-qtwebengine-5.10.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40149&attempt_id=0e1b769d-e8ed-4466-b6c2-e403b14b0642)

Attempted: qt5.qtwebengine

<details><summary>Partial log (click to expand)</summary><p>

```
[4741/17529] CXX obj/third_party/icu/bundled_icuuc/ushape.o
[4742/17529] CXX obj/third_party/icu/bundled_icuuc/ucnvdisp.o
[4743/17529] CXX obj/third_party/icu/bundled_icui18n/csrmbcs.o
[4744/17529] CXX obj/third_party/icu/bundled_icui18n/inputext.o
[4745/17529] CXX obj/third_party/icu/bundled_icuuc/utf_impl.o
[4746/17529] CXX obj/third_party/icu/bundled_icui18n/utf8collationiterator.o
[4747/17529] CXX obj/third_party/icu/bundled_icui18n/udatpg.o
[4748/17529] CXX obj/third_party/icu/bundled_icuuc/ubiditransform.o
building of '/nix/store/spbf5p9f8kg05rkdihn97qypx6pz4mr1-qtwebengine-5.10.1.drv' timed out after 3600 seconds
[31;1merror:[0m build of '/nix/store/spbf5p9f8kg05rkdihn97qypx6pz4mr1-qtwebengine-5.10.1.drv' failed
```
</p></details>

knedlsepp commented:
I guess we need something along those lines:
https://github.com/NixOS/nixpkgs/blob/d283368d73482886808b87453b838e1e85a392db/pkgs/os-specific/darwin/apple-source-releases/libsecurity_filedb/default.nix#L8-L12

knedlsepp committed:
WIP - qt5.qtwebengine: fix darwin build

knedlsepp commented:
Now it's stuck on:
```
../../3rdparty/chromium/printing/backend/cups_helper.h:8:10: fatal error: 'cups/cups.h' file not found
#include <cups/cups.h>
```

knedlsepp committed:
qt5.qtwebengine: add cups dependency

bkchr commented:
Is there any update? Does this work now?

knedlsepp commented:
It seems I got ahead of myself with this PR; This is sadly still only WIP.

Ekleog commented:
(triage) @knedlsepp, are you still working on this?

knedlsepp commented:
Not at the moment, sry. I'm gonna close this for now until I find time to actually fix it.

veprbl commented:
#53848
